smarts/core/agent_manager.py

```diff
@@ -348,15 +351,46 @@ def _filter_social_agent_actions_for_controlled_vehicles(
 
         return social_agent_actions
 
-    def send_observations_to_social_agents(self, observations):
-        """Forwards observations to managed social agents."""
-        # TODO: Don't send observations (or receive actions) from agents that have done
-        #       vehicles.
+    def add_social_observation_callback(self, callback, id):
```

Comment:
I'm not sure I like the terminology "social observation" because we already have confusion throughout SMARTS between "social _vehicles_" and "social _agents_".